test/test_other.py

```diff
@@ -14196,10 +14196,26 @@ def test_unused_destructor(self):
     self.assertNotIn(b'hello from dtor', read_binary('test_unused_destructor.wasm'))
 
   def test_strip_all(self):
-    # Test that even with `-Wl,--strip-all` the target features section is generated
-    # by wasm-ld so that later phases (e.g. wasm-opt) can read it.
-    self.do_runf('hello_world.c', emcc_args=['-Wl,--strip-all', '-pthread'])
-    self.do_runf('hello_world.c', emcc_args=['-Wl,-s', '-pthread'])
+    def has_debug_section(wasm):
+      with webassembly.Module('hello_world.wasm') as wasm:
+        return wasm.get_custom_section('.debug_info') is not None
+
+    # First, verify that `-g` produces a debug section
+    self.do_runf('hello_world.c', emcc_args=['-g'])
+    self.assertTrue(has_debug_section('hello_world.wasm'))
+
+    # Test `-Wl,--strip-all` will strip the debug section, but that the
+    # the target features section is preserved so that later phases
+    # (e.g. wasm-opt) can read it.
```

Comment:
This may have been an issue with the test before this PR, but how is this testing the features section? wasm-opt is not run normally. I think we should add `-O2` here perhaps?

Reply:
Without the `--keep-section=target_features` this test currently fails with `assert '--enable-threads' in settings.BINARYEN_FEATURES` after calling `wasm-emscripten-finalize`.

If I remove that assert it then goes onto fail at runtime because `wasm-emscripten-finalize` doesn't see the correct feature list.

Reply:
i.e. the problem isn't just wasm-opt but also `wasm-emscripten-finalize`

Reply:
I see, thanks.

Do we never skip finalize when pthreads are enabled?

Reply:
We run finalize whenever WASM_BIGINT is not enabled, regardless of `-pthread`.

Reply:
I see. lgtm then, but maybe we should add `-sWASM_BIGINT=0` here to be sure, as eventually we will flip that setting to on, and would have no way to know that this test needs an update at that time.

Reply:
I added `-O2` to be more explicit about running `wasm-opt` and/or finalize.